store/sqlstore/thread_store.go

```diff
@@ -321,19 +331,38 @@ func (s *SqlThreadStore) CollectThreadsWithNewerReplies(userId string, channelId
 	return changedThreads, nil
 }
 
-func (s *SqlThreadStore) UpdateUnreadsByChannel(userId string, changedThreads []string, timestamp int64) error {
+func (s *SqlThreadStore) UpdateUnreadsByChannel(userId string, changedThreads []string, timestamp int64, updateViewedTimestamp bool) error {
 	if len(changedThreads) == 0 {
 		return nil
 	}
-	updateQuery, updateArgs, _ := s.getQueryBuilder().
+
+	qb := s.getQueryBuilder().
 		Update("ThreadMemberships").
 		Where(sq.Eq{"UserId": userId, "PostId": changedThreads}).
-		Set("LastUpdated", timestamp).
-		Set("LastViewed", timestamp).
-		ToSql()
+		Set("LastUpdated", timestamp)
+
+	if updateViewedTimestamp {
+		qb = qb.Set("LastViewed", timestamp)
+	}
+	updateQuery, updateArgs, _ := qb.ToSql()
+
 	if _, err := s.GetMaster().Exec(updateQuery, updateArgs...); err != nil {
 		return errors.Wrap(err, "failed to update thread membership")
 	}
 
 	return nil
 }
+
+func (s *SqlThreadStore) GetPosts(threadId string, since int64) ([]*model.Post, error) {
+	query, args, _ := s.getQueryBuilder().
+		Select("*").
+		From("Posts").
+		Where(sq.Eq{"RootId": threadId}).
+		Where(sq.Eq{"DeleteAt": 0}).
+		Where(sq.GtOrEq{"UpdateAt": since}).ToSql()
+	var result []*model.Post
+	if _, err := s.GetMaster().Select(&result, query, args...); err != nil {
```

Comment:
Unless there's a strong reason not to, I'd prefer using a replica rather than the main db writer.